galera/src/replicator_str.cpp

```diff
@@ -1838,8 +1838,12 @@ void ReplicatorSMM::ist_cc(const gcs_action& act, bool must_apply,
              * Order of these calls is essential: trx_params_.version_ may
              * be altered by establish_protocol_versions() */
             establish_protocol_versions(conf.repl_proto_ver);
+            /* KH: do not reset NBO waiters. It may be that this node re-joins
```

Comment:
We always did. NBO-related writesets are stored in GCache and are transferred during IST. 
This ensures that the node which was temporarily partitioned because of a network issues can rejoin the cluster with IST and catch up with all nbo signalling (nbo end writesets) sent by other nodes to the replication channel.